### PR TITLE
Correctly scope Swift language for each test spec

### PR DESCRIPTION
**Pull request by dnkoutso**
This can wait for 1.6.0.beta.1 to ship.

### CODE REVIEW

File: spec/unit/target/pod_target_spec.rb [hunk 1]
```diff
@@ -576,95 +572,45 @@ module Pod
...
-          test_fa.stubs(:resource_bundles).returns({})
-          test_fa.stubs(:resources).returns([Pathname.new('TestModel.xcdatamodeld')])
-          @test_pod_target.stubs(:file_accessors).returns([fa, test_fa])
-          @test_pod_target.resource_paths.should == {
-            'CoconutLib' => ['${PODS_ROOT}/Model.xcdatamodeld'],
-            'CoconutLib/Tests' => ['${PODS_ROOT}/TestModel.xcdatamodeld'],
-          }
-        end
-
-        it 'returns resource paths from all specifications by default' do
```
Comment (dnkoutso) [inline spec/unit/target/pod_target_spec.rb]: This test and the above were actually duplicated!

File: spec/unit/target/pod_target_spec.rb [hunk 2]
```diff
@@ -576,95 +572,45 @@ module Pod
...
+            'WatermelonLib/SnapshotTests' => [],
           }
         end
 
-        it 'excludes framework paths from test specifications when not requested' do
-          fa = Sandbox::FileAccessor.new(nil, @coconut_spec.consumer(@platform))
-          fa.stubs(:vendored_dynamic_artifacts).returns([config.sandbox.root + Pathname.new('Vendored/Vendored.framework')])
-          @test_pod_target.stubs(:file_accessors).returns([fa])
-          @test_pod_target.stubs(:should_build?).returns(true)
+        it 'returns the correct framework paths' do
```
Comment (dnkoutso) [inline spec/unit/target/pod_target_spec.rb]: no more stubbing using the fixture properly returns the paths.

File: spec/unit/target/pod_target_spec.rb [hunk 3]
```diff
@@ -576,95 +572,45 @@ module Pod
...
-              { :name => 'Vendored.framework',
-                :input_path => '${PODS_ROOT}/Vendored/Vendored.framework',
-                :output_path => '${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/Vendored.framework',
+            'WatermelonLib' => [
+              { :name => 'WatermelonLib.framework',
+                :input_path => '${BUILT_PRODUCTS_DIR}/WatermelonLib/WatermelonLib.framework',
+                :output_path => '${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/WatermelonLib.framework',
                 },
             ],
+            'WatermelonLib/Tests' => [],
```
Comment (dnkoutso) [inline spec/unit/target/pod_target_spec.rb]: yes those are supposed to be empty.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Correctly scope Swift language for each test spec  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
No changelog entry here it is part of the one-to-one test specs feature

File: lib/cocoapods/target/pod_target.rb [hunk 4]
```diff
@@ -177,18 +177,18 @@ def uses_swift?
...
     #
-    def uses_swift_for_test_type?(test_type)
+    def uses_swift_for_test_spec?(test_spec)
       @uses_swift_for_test_type ||= {}
-      return @uses_swift_for_test_type[test_type] if @uses_swift_for_test_type.key?(test_type)
-      @uses_swift_for_test_type[test_type] = begin
-        file_accessors.select { |a| a.spec.test_specification? && a.spec.test_type == test_type }.any? do |file_accessor|
+      return @uses_swift_for_test_type[test_spec.name] if @uses_swift_for_test_type.key?(test_spec.name)
+      @uses_swift_for_test_type[test_spec.name] = begin
+        file_accessors.select { |a| a.spec.test_specification? && a.spec == test_spec }.any? do |file_accessor|
```
Comment (dnkoutso) [inline lib/cocoapods/target/pod_target.rb]: @segiddins is `a.spec == test_spec` a good check?

File: lib/cocoapods/target/pod_target.rb [hunk 5]
```diff
@@ -244,12 +244,6 @@ def contains_test_specifications?
       !test_specs.empty?
     end
 
-    # @return [Array<Symbol>] All of the test supported types within this target.
-    #
-    def supported_test_types
```
Comment (dnkoutso) [inline lib/cocoapods/target/pod_target.rb]: unused method

**Comment by segiddins** [comment]
👍🏻

**Pull request merged**